source/AsteroidField.cpp

```diff
@@ -125,9 +125,9 @@ void AsteroidField::Draw(DrawList &draw, const Point &center, double zoom) const
 
 
 // Check if the given projectile collides with any asteroids. This excludes minables.
-const vector<Collision> &AsteroidField::CollideAsteroids(const Projectile &projectile) const
+const vector<Collision> AsteroidField::CollideAsteroids(const Projectile &projectile) const
 {
-	result.clear();
+	vector<Collision> result;
```

Comment:
actually if we do it like this I would like the result vector be passed as a reference, to avoid the copy

Reply:
A reference to the vector within this function? Wouldn't that have issues with the lifetime of the vector?

Reply:
no passing a reference from the scope that is calling it

Reply:
```c++
std::vector<Collision> result;
Collide(otherArgument, result);
```

Reply:
This actually eliminates a lot of copies that were already in the code. Really nice!